capellambse/model/common/accessors.py

```diff
@@ -610,15 +616,18 @@ def __get__(self, obj, objtype=None):
         return self._make_list(obj, elems)
 
     def __set__(self, obj, value):
-        if self.aslist is not None:
-            self.__get__(obj)[:] = value
-            return
-
+        if self.aslist is None:
+            if isinstance(value, cabc.Iterable):
+                raise TypeError(f"{self._qualname} expects a single element")
```

Comment:
Doesn't this fire if value is a string (and `self.aslist is None`?

Reply:
I'm not sure if a `str` is even valid as `value` here in the first place, but even if not, we should guard against it anyways. (This stuff is annoying. Thanks for looking out for it too.)